### ISSUE TITLE
Log parser is broken for Heroku Postgres

### ISSUE THREAD

**Issue by jeffblake** [DataDog/heroku-buildpack-datadog#238]
While not directly related to the buildpack, thought posting here would get more visibility.

In Logs > Confgure > Pipelines > Heroku parser > Parsing Heroku Router and Metric logs

The last entry for Heroku postgres metrics is incorrect. 

Here is a sample log line that Heroku sends: 
`source=HEROKU_POSTGRESQL_AMBER addon=postgresql-contoured-74178 sample#current_transaction=32230410 sample#db_size=29038814063bytes sample#tables=287 sample#active-connections=40 sample#waiting-connections=0 sample#index-cache-hit-rate=0.99836 sample#table-cache-hit-rate=0.9938 sample#load-avg-1m=0.01 sample#load-avg-5m=0.02 sample#load-avg-15m=0.04 sample#read-iops=0.4477 sample#write-iops=8.8243 sample#tmp-disk-used=33849344 sample#tmp-disk-available=72944943104 sample#memory-total=15657124kB sample#memory-free=184676kB sample#memory-cached=14576644kB sample#memory-postgres=280724kB sample#wal-percentage-used=0.06858056155559547
`
Note the addition of more metrics compared to the sample line in the Grok Parser, and lack of timestamp at the beginning. Would be great to get this updated to support the additional metrics. Thanks!

**Comment by jeffblake** [comment]
I fixed the pipeline myself. Clone the standard one and use this

```
heroku.router %{_status}(?> %{_error_code} %{_error_desc})? %{_method} %{_url} %{_server_name} %{_request_id} %{_client_ip} %{_dyno} %{_time_backend_connect} %{_response_time} %{_status_code} %{_bytes_write} %{_scheme}.*

heroku.cpu_metrics (?>%{_hostname} %{_dyno}) %{_cpu_1m} %{_cpu_5m} %{_cpu_15m}.*

heroku.memory_metrics (?>%{_hostname} %{_dyno}) %{_mem_total} %{_mem_rss} %{_mem_cache} %{_mem_swap} %{notSpace} %{notSpace} %{_mem_quota}.*

heroku.postgres_metrics %{_hostname} addon=%{notSpace:heroku.addon} %{_db_current_transaction} %{_db_size} %{_db_tables} %{_db_active_connections} %{_db_waiting_connections} %{_db_index_cache_hit_rate} %{_db_table_cache_hit_rate} %{_cpu_1m} %{_cpu_5m} %{_cpu_15m} %{_iops_read} %{_iops_write} %{_db_tmp_disk_used} %{_db_tmp_disk_available} %{_mem_total} %{_mem_free} %{_mem_cache} %{_db_memory_postgres} %{_db_wal_percentage_used}.*

fallback %{data::keyvalue("=","/")}
```

and the helper rules:

```
_dyno dyno=%{notSpace:dyno}
_error_code code=%{notSpace:error.kind}
_error_desc desc="%{data:error.msg}"
_bytes_write bytes=%{integer:network.bytes_written}
_client_ip fwd="%{ipOrHost:network.client.ip}%{regex("[^\"]*")}"
_url path="%{notSpace:http.url}"
_hostname source=%{ipOrHost:syslog.hostname}
_request_id request_id=%{notSpace:http.request_id}
_response_time service=(%{integer:duration:scale(1000000)}ms|%{integer:duration:scale(1000000000)}s)
_server_name host=%{ipOrHost:http.server.name}
_status at=%{notSpace:status}
_status_code status=%{integer:http.status_code}
_time_backend_connect connect=(%{integer:load_balancer.time_backend_connect}%{notSpace:load_balancer.connect_time_unit})?
_method method=%{word:http.method}
_scheme protocol=%{word:http.url_details.scheme}
_cpu_1m sample#load(_|-)avg(_|-)1m=%{number:heroku.cpu["1m"]:scale(100)}
_cpu_5m sample#load(_|-)avg(_|-)5m=%{number:heroku.cpu["5m"]:scale(100)}
_cpu_15m sample#load(_|-)avg(_|-)15m=%{number:heroku.cpu["15m"]:scale(100)}
_mem_total sample#memory(_|-)total=(%{number:heroku.memory.total:scale(1000000)}MB|%{number:heroku.memory.total:scale(1000)}kB)
_mem_free sample#memory-free=%{number:heroku.memory.free:scale(1000)}kB
_mem_rss sample#memory_rss=%{number:heroku.memory.rss:scale(1000000)}MB
_mem_cache sample#memory(_|-)cache(d)?=(%{number:heroku.memory.cache:scale(1000000)}MB|%{number:heroku.memory.cache:scale(1000)}kB)
_mem_swap sample#memory_swap=%{number:heroku.memory.swap:scale(1000000)}MB
_mem_pgpgin sample#memory_pgpgin=%{number:heroku.memory.pgpgin}%{number:heroku.memory.pgpgin_unit}
_mem_pgpgout sample#memory_pgpgin=%{number:heroku.memory.pgpout}%{number:heroku.memory.pgpgout_unit}
_mem_quota sample#memory_quota=%{number:heroku.memory.quota:scale(1000000)}MB
_db_size sample#db_size=%{number:heroku.db.size}bytes
_db_tables sample#tables=%{number:heroku.db.tables}
_db_active_connections sample#active-connections=%{number:heroku.db.active_connections}
_db_current_transaction sample#current_transaction=%{notSpace:heroku.db.transaction_id}
_db_index_cache_hit_rate sample#index-cache-hit-rate=%{number:heroku.db.cache_hit_rate.index}
_db_table_cache_hit_rate sample#table-cache-hit-rate=%{number:heroku.db.cache_hit_rate.table}
_db_waiting_connections sample#waiting-connections=%{number:heroku.db.waiting_connections}
_db_memory_postgres sample#memory-postgres=%{number:heroku.db.memory:scale(1000)}kB
_db_follower_lag_commits sample#follower-lag-commits=%{number:heroku.db.follower_lag_commits}
_iops_read sample#read-iops=%{number:heroku.iops.read}
_iops_write sample#write-iops=%{number:heroku.iops.write}

_timestamp %{date("yyyy-MM-dd'T'HH:mm:ssZ"):date}
_db_wal_percentage_used sample#wal-percentage-used=%{number:heroku.db.wal_percentage_used:scale(100)}
_db_tmp_disk_used sample#tmp-disk-used=%{number:heroku.db.tmp_disk_used:scale(1000)}
_db_tmp_disk_available sample#tmp-disk-available=%{number:heroku.db.tmp_disk_available:scale(1000)}

```

**Comment by carltondickson** [comment]
Thanks for posting this @jeffblake as I'm trying to get my postgres logs into DataDog and parsed..I can see them in the live tail but only the full log message (as can be seen in Heroku logs) and it doesn't look like it's being parsed correctly.

Not sure if it's a thing but we had to clone the Heroku pipeline some time ago to do some log remapping I think it was.

I've just installed the postgres integration but noticed `heroku.postgres_metrics...` was missing from our "Grok Parser: Parsing Heroku Router and Metric logs".
